File: frontend/src/modules/auth/services/roleService.ts
import { apiDelete, apiGet, apiPost, apiPut } from '@/services/apiClient';
import type { RoleRequest, RoleResponse } from '../types';

/** Backend: auth/controller/RoleController.java */
export const roleService = {
  list: () => apiGet<RoleResponse[]>('/v1/roles'),

  get: (id: number) => apiGet<RoleResponse>(`/v1/roles/${id}`),

  create: (body: RoleRequest) => apiPost<RoleResponse>('/v1/roles', body),

  update: (id: number, body: RoleRequest) => apiPut<RoleResponse>(`/v1/roles/${id}`, body),

  remove: (id: number) => apiDelete(`/v1/roles/${id}`),
};

File: backend/src/main/java/com/hardware/erp/auth/service/RoleService.java
package com.hardware.erp.auth.service;

import com.hardware.erp.auth.dto.RoleRequest;
import com.hardware.erp.auth.dto.RoleResponse;

import java.util.List;

public interface RoleService {

    List<RoleResponse> findAll();

    RoleResponse get(Long id);

    RoleResponse create(RoleRequest request);

    RoleResponse update(Long id, RoleRequest request);

    void delete(Long id);
}
